decoder.go

```diff
@@ -35,7 +46,19 @@ func Unmarshal(bs []byte, v interface{}) error {
 // Decode reads the given input and decodes it into the structure
 // given by `out`.
 func Decode(out interface{}, in string) error {
-	obj, err := Parse(in)
+	return decode(out, in, false)
+}
+
+// Decode reads the given input and decodes it into the structure
+// given by `out`.
+func DecodeErrorOnDuplicates(out interface{}, in string) error {
+	return decode(out, in, true)
+}
+
+// Decode reads the given input and decodes it into the structure
+// given by `out`.
+func decode(out interface{}, in string, errorOnDuplicateAtributes bool) error {
```

Comment:
very very nit: the function descriptions for `Decode` `DecodeErrorOnDuplicates` and `decode` are identical, maybe give a small difference to make it clearer.